src/Core/Json/JsonSerializer.php

```
@@ -0,0 +1,192 @@
+    /**
+     * Serializes an array based on type annotations (either a list or map).
+     *
+     * @param mixed[]|array<string, mixed> $data The array to be serialized.
+     * @param mixed[]|array<string, mixed> $type The type definition from the annotation.
+     * @return mixed[]|array<string, mixed> The serialized array.
+     * @throws JsonException If serialization fails.
+     */
+    public static function serializeArray(array $data, array $type): array
+    {
+        return Utils::isMapType($type)
+            ? self::serializeMap($data, $type)
```

PHPStan (2.1.17)

Parameter #1 $data of static method Intercom\Core\Json\JsonSerializer::serializeMap() expects array<string, mixed>, array<mixed> given.


(argument.type)